README.md

```diff
@@ -1,655 +1,132 @@
-
-# Add the gem shopify_app to your Gemfile
-$ bundle add shopify_app
-```
+You can find documentation on gem usage, concepts, mixins, installation, and more in [`/docs`](/docs).
 
-Now we are ready to run any of the [generators](#generators) included with `shopify_app`. The following section explains the generators and what you can do with them.
+* Start with the [*Generators*](/docs/shopify_app/generators.md) document to learn more about the generators this gem offers.
+* Check out the [*Changelog*](/CHANGELOG.md) for notes on the latest gem releases.
+* See [*Troubleshooting*](/docs/Troubleshooting.md) for tips on common issues.
+* If you are looking to upgrade your Shopify App version to a new major release, see [*Upgrading*](/docs/Upgrading.md) for important notes on breaking changes.
 
+### Overview
 
-#### Rails Compatibility
+[Quickstart](/docs/Quickstart.md)
 
-The latest version of shopify_app is compatible with Rails `>= 5`. Use version `<= v7.2.8` if you need to work with Rails 4.
+[Releasing](/docs/Releasing.md)
 
+[Troubleshooting](/docs/Troubleshooting.md)
 
-Generators
-----------
-
-### API Keys
-<!-- This anchor name `#api-keys` is linked to from user output in `templates/shopify_app.rb.tt` so beware of changing -->
-Before starting the app, you'll need to ensure it can read the Shopify environment variables `SHOPIFY_API_KEY` and `SHOPIFY_API_SECRET`.
-
-In a development environment, a common approach is to use the [dotenv-rails](https://github.com/bkeepers/dotenv) gem, along with an `.env` file in the following format:
-
-```
-SHOPIFY_API_KEY=your api key
-SHOPIFY_API_SECRET=your api secret
-```
-
-These values can be found on the "App Setup" page in the [Shopify Partners Dashboard][dashboard].
-(If you are using [shopify-app-cli](https://github.com/Shopify/shopify-app-cli) this `.env` file will be created automatically).
-If you are checking your code into a code repository, ensure your `.gitignore` prevents your `.env` file from being checked into any publicly accessible code.
-
-### Default Generator
-
-The default generator will run the `install`, `shop`, `authenticated_controller`, and `home_controller` generators. This is the recommended way to start a new app from scratch:
-
-```sh
-$ rails generate shopify_app
-```
-
-After running the generator, you will need to run `rails db:migrate` to add new tables to your database. You can start your app with `bundle exec rails server` and install your app by visiting `http://localhost` in your web browser.
+[Upgrading](/docs/Upgrading.md)
 
+[shopify_app](/docs/shopify_app)
+  * [Authentication](/docs/shopify_app/authentication.md)
+  * [Engine](/docs/shopify_app/engine.md)
+  * [Generators](/docs/shopify_app/generators.md)
+  * [ScriptTags](/docs/shopify_app/script-tags.md)
+  * [Session repository](/docs/shopify_app/session-repository.md)
+  * [Testing](/docs/shopify_app/testing.md)
+  * [Webhooks](/docs/shopify_app/webhooks.md)
 
-### Install Generator
+### Engine
```

Comment:
I wonder if it makes more sense for this section to be here or under the Engine page. It is pretty relevant stuff but I wonder if it will confuse people to have engine docs in two places.

Reply:
I'd like to leave this information in the README as a digestible table.

My opinion is that partners shouldn't dig to discover these core routes. But, more detail that is relevant to this engine should be available as a separate doc.

Reply:
Yes, I had the same impression - the table makes it very clear how the engine affects the app. Let's keep it.